### PR TITLE
update deps on external client pkgs

### PR DESCRIPTION
**Pull request by aruniverse**
- This pr initially started off as removing an unnecessary dep on itwin/browser-authorization in appui-react which caused an older ver of core to get pulled in for consumers

- Updated auth, project, imodels, and reality-data client pkgs to ver that uses official 3.0 release deps

- Ran into some typescript compilation issues with union types in core-backend and -transformer, updated those accordingly

Need to backport/cherrypick this to master

### CODE REVIEW

**Comment by aruniverse** [comment]
Not sure why its failing to build core-backend all of a sudden, 
image

**Review by calebmshafer — APPROVED** [review]

**Comment by pmconne** [comment]
@aruniverse for some reason when I pulled your branch there were conflicts in browser-approved-packages and pnpm-lock. I did a rush update and successfully built core-backend.

**Review by MarcBedard8 — APPROVED** [review]

File: common/changes/@itwin/physical-material-backend/update-deps_2022-01-25-17-52.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/physical-material-backend",
```
Comment (calebmshafer) [inline common/changes/@itwin/physical-material-backend/update-deps_2022-01-25-17-52.json:4]: I think you accidentally ran this without the branch specified. Running `rush change -b origin/release/3.0.x` should fix it.

There is an ongoing issue with our rush change python script that @skirby1996 is looking into that prevents any PRs from currently being completed.

Comment (pmconne) [reply common/changes/@itwin/physical-material-backend/update-deps_2022-01-25-17-52.json:4]: Yeah that was me...i'll fix.

Comment (pmconne) [reply common/changes/@itwin/physical-material-backend/update-deps_2022-01-25-17-52.json]: fixed.

**Comment by aruniverse** [comment]
> @aruniverse when I pulled your branch there were conflicts in browser-approved-packages and pnpm-lock. I did a rush update and successfully built core-backend.

thanks paul; not sure how my branch was out of date. I'm still running into the core-backend build error locally and in the ci. ill continue to investigate why.

**Review by ColinKerr — APPROVED** [review]

**Comment by calebmshafer** [comment]
@Mergifyio backport master

**Comment by calebmshafer** [comment]
@bsteinbk mind taking a look?

**Review by bsteinbk — APPROVED** [review]

**Review by pmconne — APPROVED** [review]

**Review by MichaelBelousov — APPROVED** [review]

File: common/config/rush/.pnpmfile.cjs [hunk 2]
```diff
@@ -1,20 +1,23 @@
 function readPackage(pkg) {
   if ((pkg.name == "typedoc" || pkg.name == "@microsoft/api-extractor") && pkg.dependencies && pkg.dependencies["typescript"])
     pkg.dependencies["typescript"] = "~4.4.0";
+  // Hacky mess: For external packages to this monorepo that have peer dependencies on packages
```
Comment (aruniverse) [inline common/config/rush/.pnpmfile.cjs:4]: im surprised we dont need to do this for the reality-data, projects, browser-auth clients
@calebmshafer

Comment (calebmshafer) [reply common/config/rush/.pnpmfile.cjs:4]: I think it's due to they're just type dependencies in those packages and if you look in the lock file you'd find the same issue.

**Comment by calebmshafer** [comment]
@aruniverse, @pbell97 fixed the build error on master but didn't backport it to the 3.0.x branch. Here's the backport PR, #3140.

**Comment by calebmshafer** [comment]
@grigasp can you review/approve?

**Review by saskliutas — APPROVED** [review]

**Pull request merged**